appeals/web/src/server/lib/page-components/document.js

```diff
@@ -0,0 +1,64 @@
+import { isFolderInfo } from '#lib/ts-utilities.js';
+import { mapActionComponent } from '#lib/mappers/component-permissions.mapper.js';
+import { formatDocumentActionLink, formatDocumentValues } from '#lib/display-page-formatter.js';
+
+/**
+ * Returns the instructions for a document display field
+ *
+ * @param {Object} options
+ * @param {string} options.id
+ * @param {string} options.text
+ * @param {number} options.appealId
+ * @param {import('@pins/appeals.api').Appeals.FolderInfo|null|undefined} options.folderInfo
+ * @param {string} options.requiredPermissionName
+ * @param {import('../../app/auth/auth-session.service.js').SessionWithAuth} options.session
+ * @param {string} options.manageUrl
+ * @param {string} options.uploadUrlTemplate
+ * @param {string} [options.cypressDataName]
+ * @returns {Instructions}
+ */
+export function documentTypeDisplayField({
+	id,
```

Comment:
Just FYI there's a tech debt ticket here to refactor the shared document code to use this destructured object pattern for parameters. If it's been decided this approach should be used for all new code, should we extend that to cover all existing functions in the codebase? https://pins-ds.atlassian.net/browse/A2-949

Reply:
Agreed. I think far-reaching changes in convention like this should be discussed or at least communicated to the team so we're aware this is what we should be doing when writing new code. Also we should open a tech debt ticket to ensure existing code not using this approach is refactored, or we'll end up with inconsistency, which is arguably worse than one consistent non-ideal approach

Reply:
Yes there is always a trade off between going in a new direction and having differing approaches, or sticking with what is there. I think in this instance the two can co-exist without issue, because we type-check function calls. Given how much clearer it makes these functions, it makes sense to use named parameters here. We don't need to refactor the whole codebase for this case, and for many functions it won't matter. If there are some key functions that would benefit, as in that ticket, it may make sense to refactor them, but I would keep it targeted.